internal/querycoordv2/checkers/segment_checker.go

```diff
@@ -236,7 +236,13 @@ func (c *SegmentChecker) getSealedSegmentDiff(
 		_, existOnCurrent := currentTargetMap[segment.GetID()]
 		_, existOnNext := nextTargetMap[segment.GetID()]
 
-		if !existOnNext && !existOnCurrent {
+		l0WithWrongLocation := false
+		if existOnCurrent {
+			leader := c.dist.LeaderViewManager.GetLatestLeadersByReplicaShard(replica, segment.GetInsertChannel())
+			l0WithWrongLocation = segment.GetLevel() == datapb.SegmentLevel_L0 && segment.Node != leader.ID
+		}
+
+		if !existOnNext && !existOnCurrent || l0WithWrongLocation {
 			toRelease = append(toRelease, segment)
```

Comment:
shall add "l0WithWrongLocation" check in toLoad as well

Reply:
otherwise the right l0 segment could never be loaded since toRelease will be filtered by `leaderViewInUse` condition

Reply:
wanted logic already exist
![image](https://github.com/milvus-io/milvus/assets/108661493/9872efb7-ae1d-43be-be00-309eab4f13a1)